dynamic_programming/all_construct.py

```diff
@@ -0,0 +1,56 @@
+"""
+Program to list all the ways a target string can be
+constructed from the given list of substrings
+"""
+
+
+def all_construct(target: str, word_bank: list[str] = None) -> list[list[str]]:
```

Comment:
`list[str] = None` makes no sense because None is not a list.
```suggestion
def all_construct(target: str, word_bank: list[str] | None = None) -> list[list[str]]:
```

Reply:
I followed the article you mentioned previously regarding mutable default arguments
[](https://docs.python-guide.org/writing/gotchas/)
and I don't see a problem in defining a default argument as None and later changing it to an empty list in the code.

Reply:
Ohh I think I see.
 do you mean the type hint is not clear that way?